purpleair-aqi.js

```diff
@@ -134,38 +134,32 @@ async function getSensorId() {
     const seLat = latitude - BOUND_OFFSET;
     const nwLng = longitude - BOUND_OFFSET;
     const seLng = longitude + BOUND_OFFSET;
-    const req = new Request(
-      `${API_URL}/v1/sensors?fields=name,latitude,longitude,location_type&max_age=3600&location_type=0&nwlat=${nwLat}&selat=${seLat}&nwlng=${nwLng}&selng=${seLng}`
+    var req = new Request(
+      `${API_URL}/v1/sensors?fields=name,latitude,longitude&max_age=3600&location_type=0&nwlat=${nwLat}&selat=${seLat}&nwlng=${nwLng}&selng=${seLng}`
     );
+    req.headers = {"X-API-Key": API_KEY} ;
 
-    /** @type {{ code?: number; results?: Array<Object<string, number|string>>; }} */
     const res = await req.loadJSON();
-
-    const { results } = res;
-
-    const sensorIdField = "ID";
-    const latField = "Lat";
-    const lonField = "Lon";
-    const locationField = "DEVICE_LOCATIONTYPE";
-    const ageField = "AGE";
-    const OUTDOOR = "outside";
+    const { data } = res;
 
     let closestSensor;
     let closestDistance = Infinity;
 
-    for (const location of results.filter((datum) => datum[locationField] === OUTDOOR && datum[ageField] < 60 * 4)) {
+    for (const location of data) {
       const distanceFromLocation = haversine(
         { latitude, longitude },
-        { latitude: location[latField], longitude: location[lonField] }
+        { latitude: location[2], longitude: location[3] }
       );
       if (distanceFromLocation < closestDistance) {
         closestDistance = distanceFromLocation;
         closestSensor = location;
       }
     }
 
-    const id = closestSensor[sensorIdField];
-    cacheData("sensor.json", { id, updatedAt: Date.now() });
+    const [id, name, lat, long] = closestSensor;
```

Comment:
This "destructures" the array into the 4 fields, which I then use to to create an easier to use object.